### PR TITLE
Hourly Weather View

### PR DESCRIPTION
**Pull request by redbeed**
Hello,

with "show (me) weather" you can display the weather for the next 20 hours.
Also i add an string formatter to add an whitespace for pos. temps.

### CODE REVIEW

File: js/weather-service.js [hunk 1]
```diff
@@ -25,7 +28,7 @@
                 return null;
             }
             service.forcast.data.currently.day = moment.unix(service.forcast.data.currently.time).format('ddd')
-            service.forcast.data.currently.temperature = parseFloat(service.forcast.data.currently.temperature).toFixed(1);
+            service.forcast.data.currently.temperature = vsprintf(numberFormat, [parseFloat(service.forcast.data.currently.temperature).toFixed(1)]);
```
Comment (evancohen) [inline js/weather-service.js]: Won't `vsprintf` will be undefined here?

File: js/controller.js [hunk 2]
```diff
@@ -266,6 +266,12 @@
                 $scope.focus = "dilbert";
             });
 
+            //show hourly weather
+            AnnyangService.addCommand('Show (the) weather', function() {
```
Comment (evancohen) [inline js/controller.js]: Should be
``` javascript
AnnyangService.addCommand('Show (me)(the) weather', function() {
```
Which will match
- show weather
- show me weather
- show the weather
- show me the weather

Comment (redbeed) [inline js/controller.js]: Yes, but we have a Show me... commands. So its not possible

File: js/weather-service.js (hunk 1 above)
Comment (redbeed) [inline js/weather-service.js]: load the js-file in index.html and this create the function.

Comment (evancohen) [inline js/weather-service.js]: This seems like a pointless dependency to me...
Let's say `service.forcast.data.currently.temperature` = 123.45
``` javascript
parseFloat(service.forcast.data.currently.temperature).toFixed(1); //123.5
vsprintf(numberFormat, [parseFloat(service.forcast.data.currently.temperature).toFixed(1)]) //123.5
```

Why do we need this?

Comment (redbeed) [inline js/weather-service.js]: The format is to add a whitespace if no "-" as First char.(only for Numbers with 4 chars)

Comment (evancohen) [inline js/weather-service.js]: This is something that you could do with CSS though.

Comment (redbeed) [inline js/weather-service.js]: mhh.. yes i will try it.

**Comment by evancohen** [comment]
@kurtdb @joelhawksley this is a pretty stale PR. Is this something that we should bring up to date and take? The weather UI has improved pretty significantly here, so I'm not sure if this is still relevant.

**Comment by joelhawksley** [comment]
@evancohen I don't any strong feelings here. I'd leave it up to the author, @redbeed to either update the PR or close it in the next week, as it has not been touched in > 4months.

**Comment by evancohen** [comment]
Sounds good. @redbeed, thanks for taking the time to put this together, since this PR is really out of date we're going to close it. Feel free to update and reopen or send another PR.

**Pull request closed without merging**